### ISSUE TITLE
Notepad++ opens slowly with many files

### ISSUE THREAD

**Issue by BullyWiiPlaza** [notepad-plus-plus/notepad-plus-plus#14000]
### Description of the Issue
Over time, I opened up many files in the Notepad++ tabs to work with (e.g. 270 or so) and I don't bother closing them all one of one. When I launch Notepad++ from scratch, it takes quite a long time to load.

### Steps to Reproduce the Issue
1. Open Notepad++
2. Open tons of random text files, including I guess large files (with a couple MB)
3. Close Notepad++
4. Open Notepad++
5. The launch will take a long time

### Expected Behavior
I expect Notepad++ to be smarter and more efficient about opening/loading/rendering remembered files so it can start up very fast and be usable very fast. Currently, this is way too slow (30s delay?). If lets say I have 10 tabs on the screen, why are all other files still loaded/rendered in the background (?) before the editor becomes responsive? If there are a couple large files included, it would be very inefficient to process everything on application launch and I wasn't going to use those files anyway. What about [lazy loading](https://www.imperva.com/learn/performance/lazy-loading/#:~:text=Lazy%20loading%20is%20the%20practice,performance%20and%20save%20system%20resources.) or multi-threading to optimize this? Again, I didn't study the exact implementation, this is just guesswork of what could be the problem.

### Actual Behavior
The load time of a cold start of Notepad++ is ineffficient and very slow when many files are being loaded.

### Debug Information
```
Notepad++ v8.5.4   (64-bit)
Build time : Jun 17 2023 - 20:42:45
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 Pro (64-bit)
OS Version : 21H2
OS Build : 22000.2295
Current ANSI codepage : 65001
Plugins : 
    JSMinNPP (1.2205)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
```

**Comment by michael-milette** [comment]
Hi @BullyWiiPlaza ,

I too have many open files (about 280 unsaved actually) and when I open Notepad++,  it takes a while to open all the files. Notepad++ is very good at opening files however, launch time suffers, even when running on an SSD.

I think Lazyloading the files would be fantastic. It would no longer matter how many files you previously had open. It would be great if it could open the file in the current tab first and then either load other files as you switch tabs or load the rest in a lower-priority background task.

Thank you for considering this request.

Best regards,

Michael Milette

**Comment by sochartgit** [comment]
Any update, same problem here ?

**Comment by roypa-dxb** [comment]
Delay/Slowness due to all the previos sessions, Disable this from the below settings :)

Go to -> Settings ->Preferences ->Backup->Untick Remember current session for next launch

**Comment by BullyWiiPlaza** [comment]
> Delay/Slowness due to all the previos sessions, Disable this from the below settings :)
> 
> Go to -> Settings ->Preferences ->Backup->Untick Remember current session for next launch

No, that defeats the purpose of this feature. It would be great if the loading performance was addressed.

**Comment by xenago** [comment]
Adding my experience to this. I have many files stored on network drives and was surprised that the entire application is unusable if even one of them takes a long time to load.

**Comment by sochartgit** [comment]
I've gone back to another editor which works perfectly over LAN, over WAN and locally. No more time to waste with NP++ considering no reply for devs.

**Comment by Tragen** [comment]
Same problem here. This would be amazing.
Just lazy load everything after the tabs are shown and the new file is loaded.

**Comment by marcosjose** [comment]
I’m having the same problem with NPP being slow to open, related to the fact that I have several “draft” (unsaved) files saved, since I have the options enabled: backup and remember last session.
In my case, I have an SSD and it takes about 20 seconds to load about 200 files. So it spends about 0.1 seconds per file.

I did some tests using Process Monitor, as shown in the image below.

<img width="1440" height="860" alt="Image" src="https://github.com/user-attachments/assets/c0b4499d-13ca-43d4-8b28-1adcaf6813fb" />

Regarding the Process Monitor, I followed the instructions contained on the following page:

https://community.notepad-plus-plus.org/topic/23340/notepad-very-slow-to-open/16

In my case, it seems that NPP keeps trying to create files in the “Program Files” folder, but it gives a “NAME NOT FOUND” error.
After that, it goes to the “AppData/Roaming…” folder where the backup files are located.

In short, based on this result, does anyone have any idea how to try to solve this delay, perhaps caused by failures or repeated attempts to read/write the backup files?

**Comment by softyoda** [comment]
> I've gone back to another editor which works perfectly over LAN, over WAN and locally. No more time to waste with NP++ considering no reply for devs.

What is the name of your editor ? I wished "quick look" would be able to edit simple .md or .txt files, but it [doesn't ](https://github.com/QL-Win/QuickLook/issues/532) , notepad++ is too old from old days of windows xp age, i wish people who have in mind a modern ux and way to develop software that would be efficient (with lazy loading for example) would exist, but i haven't found yet.
I also wish there were location to crowdfund such floss things.

**Comment by softyoda** [comment]
[👀](https://manifold.markets/softyoda/will-issue-14000-of-httpsgithubcomn?r=c29mdHlvZGE)

**Comment by softyoda** [comment]
This issue is a duplicate of https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11741

**Comment by softyoda** [comment]
Can i give a bounty ? I lose 50sec each time i open notepad++ and it's 2026, my workstation have leastest gen hardware

**Comment by enginelesscc** [comment]
> Can i give a bounty ? I lose 50sec each time i open notepad++ and it's 2026, my workstation have leastest gen hardware

please try my PR
